project-base/templates/Front/Content/Product/filterFormMacro.html.twig

```diff
@@ -4,6 +4,16 @@
 
         {% set isSearch = page == 'search' %}
 
+        {% set defaultOpened = true %}
+
+        {% if defaultOpened %}
+            {% set defualtTitleClass = 'active' %}
+            {% set defualtInputStyle = 'display: block;' %}
+        {% else %}
+            {% set defualtTitleClass = '' %}
+            {% set defualtInputStyle = 'display: none;' %}
+        {% endif %}
```

Comment:
Are these 6 lines of code needed here? I understand why it is here, but maybe you can do the same thing only with CSS and some class. Is this condition worth it here?

Reply:
I know what you mean. I think this is compotent state variable setting - and that shouldn´t be in css. We usually configure all component states designs in css, but not logic.

I think it is not good to go to css/less to change system configuration.

Reply:
Same as that logic should not be in twig templates :)

Reply:
Hi @henzigo 
Thx for comment. In my point of view, this block of code is about setting default CSS style after accessing a page with filter. 
By default, setting `defaultOpened` determines that filters are open. It is global setting for all pages, which use this macro. Anyone who wants to change it, can modify macro and send this setting from outside. 
I was thinking about it for a while, but for now I don't see any reason, why to set this setting as macro parameter and everywhere, where it is used send value `true`.

Reply:
I never said that you should set is as macro parameter. I have just problem with logic in twig templates. But do what you think is best :)

Reply:
I am confused from these lines.

Full context is with definiton `{% set defaultOpened = true %}` (now It is renamed 
 to `productFilterBoxesOpened`) and it is mindfuck for me. Why I have condition with else block when the condition is always met. This code in project is confused and I remove it when I see it. It is in project-base -> project

Second point is: Why inline style? We have class `display-none`...or not?

Reply:
Plus there is class `active` which can set all necessary attributes/state for whole component (not one class for title, another style/class for input)

Reply:
Condition with else - you can easily setup filter behaviour without understanding not so easy code at first look. So with this switcher you can close all filter groups easily. All these groups can be opened and closed by clicking on arrow at title - and it brings inline style. So I just use these behaviour and made one condition - so developer can easy change it.

Reply:
My point of view is: It is confusing for first contact with developer. On real project I dont want this code.

It can be easy for change, but readability/understanability is worse in my opinion. 

And inline style is about architecture/design of component. One class in component (eg.  `active`) can show/colorize title and hide/input and vice versa.

Reply:
`active` - class can do it, but it can make slide down / slide up effect. It opens and closes element immediately.

Readibility/understandability
Can you explain this for me more? There are lot off conditions in code. Some of them are evaluated by variables "from somewhere" and if you want to change something you have to find where "are variables born".

In this situation for me - you have easy if else condition - with one variable which is straight one line above. I think this is more understable than 90% other conditions.

Thank you for explaining me. (-;

Reply:
I think i finally get how you meaned thing about inline styles. But after short discussion with colleaguess I made decision to remove setup possibility - and if we need to resolve it in future, we will do it better way. For now it will be opened as default without true/false setting.